pypeit/scripts/coadd_2dspec.py

```diff
@@ -78,121 +51,109 @@ def get_parser(cls, width=None):
+        # TODO Heliocentric for coadd2d needs to be thought through. Currently turning it off.
+        if par['calibrations']['wavelengths']['refframe'] != 'observed':
+            msgs.warn('Wavelength reference frame shift (e.g., heliocentric correction) not yet '
+                      'fully developed.  Ignoring input and setting "refframe = observed".')
+            par['calibrations']['wavelengths']['refframe'] = 'observed'
+        # TODO Flexure correction for coadd2d needs to be thought through. Currently turning it off.
+        if par['flexure']['spec_method'] != 'skip':
+            msgs.warn('Spectroscopic flexure correction not yet fully developed.  Skipping.')
+            par['flexure']['spec_method'] = 'skip'
+        # TODO This is currently the default for 2d coadds, but we need a way to toggle it on/off
+        if not par['reduce']['findobj']['skip_skysub']:
+            msgs.warn('Must skip sky subtraction when finding objects (i.e., sky should have '
+                      'been subtracted during primary reduction procedure).  Skipping.')
+            par['reduce']['findobj']['skip_skysub'] = True
+
+        # Get the files
+        spec2d_files = coadd2dFile.filenames
+
+        # Get the paths
+        # NOTE: These two should be the same as 
+        #   Path(par['rdx']['redux_path']).resolve() / par['rdx']['scidir']
+        #   Path(spec2d_files[0]).parent
+        pypeit_scidir = Path(spec2d_files[0]).parent
+        coadd_scidir = pypeit_scidir.parent / f"{par['rdx']['scidir']}_coadd"
+        if not coadd_scidir.exists():
+            coadd_scidir.mkdir(parents=True)
+#        pypeit_calib_dir = pypeit_scidir.parent / par['calibrations']['calib_dir']
+#        coadd_calib_dir = pypeit_scidir.parent / f"{par['calibrations']['calib_dir']}_coadd"
+#        if not coadd_calib_dir.exists():
+#            coadd_calib_dir.mkdir(parents=True)
+#        pypeit_qa_dir = pypeit_scidir.parent / par['rdx']['qadir']
+#        coadd_qa_dir = pypeit_scidir.parent / f"{par['rdx']['qadir']}_coadd"
+#        qa_path = coadd_qa_dir / 'PNGs'
+        par['rdx']['qadir'] += '_coadd'
+        qa_path = pypeit_scidir.parent / par['rdx']['qadir'] / 'PNGs'
+        if not qa_path.exists():
+            qa_path.mkdir(parents=True)
+
+        # Get the output basename
         head2d = fits.getheader(spec2d_files[0])
         if args.basename is None:
-            #TODO Fix this, currently does not work if target names have - or _
-            filename_first = os.path.basename(spec2d_files[0])
-            filename_last = os.path.basename(spec2d_files[-1])
-            prefix_first = (filename_first.split('_')[1]).split('-')[0]
-            prefix_last = (filename_last.split('_')[1]).split('-')[0]
-            objname = (filename_first.split('-')[1]).split('_')[0]
-            basename = '{:s}-{:s}-{:s}'.format(prefix_first,prefix_last,objname)
+            lasthdr = fits.getheader(spec2d_files[-1])
+            if 'FILENAME' not in head2d:
```

Comment:
Which command line option are you referring to?

Reply:
I assume the `--basename` option, but I think I've since lost the comment...